src/common/store-proxy.ts

```diff
@@ -1,33 +1,25 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { StoreUpdateMessageDistributor } from 'common/store-update-message-distributor';
 import { isEqual } from 'lodash';
 
 import { BaseStore } from './base-store';
-import { BrowserAdapter } from './browser-adapters/browser-adapter';
 import { GenericStoreMessageTypes } from './constants/generic-store-messages-types';
 import { Store } from './flux/store';
-import { StoreType } from './types/store-type';
 import { StoreUpdateMessage } from './types/store-update-message';
 
 export class StoreProxy<TState> extends Store implements BaseStore<TState> {
     private state: TState;
-    private storeId: string;
-    private tabId?: number;
-    private browserAdapter: BrowserAdapter;
 
-    constructor(storeId: string, browserAdapter: BrowserAdapter, tabId?: number) {
+    constructor(
+        private readonly storeId: string,
+        private readonly messageDistributor: StoreUpdateMessageDistributor,
+    ) {
         super();
-        this.storeId = storeId;
-        this.browserAdapter = browserAdapter;
-        this.tabId = tabId;
-        this.browserAdapter.addListenerOnMessage(this.onChange);
+        this.messageDistributor.registerStoreUpdateListener(storeId, this.onChange);
     }
 
     private onChange = (message: StoreUpdateMessage<TState>): void => {
-        if (!this.isValidMessage(message)) {
-            return;
-        }
-
         if (
             message.messageType === GenericStoreMessageTypes.storeStateChanged &&
```

Comment:
It feels off for this to be handled by the `StoreProxy` when the new usage expectation is that the distributor has already validated it's a store update message before passing it here. It looks like this check and the `message.isStoreUpdateMessage` check in L46 of the distributor are redundant; maybe move this check over to there and update the `StoreUpdateMessage` type to drop `isStoreUpdateMessage` entirely and retype `messageType` as the constant `GenericStoreMessageTypes.storeStateChanged` instead of `string`?